### PR TITLE
Add document upload functionality

### PR DESCRIPTION
**Pull request by CrystalPea**
- [x] I’ve used the pull request template
- [x] I’ve written unit tests for these changes
- [x] I’ve updated the documentation in
  - `README.md`
  - `CHANGELOG.md`
- [x] I’ve bumped the version number in
  - `package.json`
- [x] I've added new environment variables in
  - `CONTRIBUTING.md`
  - `notifications-node-client/scripts/generate_docker_env.sh`

### CODE REVIEW

**Comment by CrystalPea** [comment]
@jonathanglassman See changes in documentation :)

File: client/notification.js [hunk 1]
```diff
@@ -291,7 +291,12 @@ _.extend(NotifyClient.prototype, {
    */
   setProxy: function(url) {
     this.apiClient.setProxy(url);
-  }
+  },
+
+  prepareUpload: function(pdf_data) {
+    if (pdf_data.length > 2 * 1024 * 1024) { throw "This file is above the 2MB limit."; }
```
Comment (servingUpAces) [inline client/notification.js]: Do we need to add this error to the Error table in the documentation?

File: spec/notification.js [hunk 2]
```diff
@@ -63,12 +63,14 @@ describe('notification api', () => {
 
     });
 
-    it('should send an email with email_reply_to_id', () => {
+    it('should send an email with document upload', () => {
```
Comment (servingUpAces) [inline spec/notification.js]: Do we want one test for send email? Or another without the document upload?

**Comment by servingUpAces** [comment]
It would be nice if we had an integration test for document upload.

File: spec/notification.js (hunk 2 above)
Comment (CrystalPea) [reply spec/notification.js]: I will re-add the previous one, so there are two.

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -250,6 +250,75 @@ If you omit this argument your default email reply-to address will be set for th
...
+
+Pass the file object as a value into the personalisation argument. For example:
+
+```javascript
+var fs = require('fs');
+
+fs.readFile('path/to/document.pdf', function(err, document) {
+	console.log(err);
+	notifyClient.sendEmail('your_notify_api_key', 'recipient@email.com', {
+		personalisation: notifyClient.prepareUpload(document),
```
Comment (leohemsted) [inline DOCUMENTATION.md]: This needs a couple of clean-ups - first arg is templateId, not api key, and personalisation should be a dictionary with the key `link_to_document`. We should keep it the same as the python client docs for consistency. So something like

```js
notifyClient.sendEmail(templateId, emailAddress, {
    personalisation: {
        first_name: 'Amala',
        application_date: '2018-01-01',
        link_to_document: notifyClient.prepareUpload(document)
    }
}
```

File: README.md [hunk 4]
```diff
@@ -250,6 +250,74 @@ If you omit this argument your default email reply-to address will be set for th
...
+
+Pass the file object as a value into the personalisation argument. For example:
+
+```javascript
+var fs = require('fs');
+
+fs.readFile('path/to/document.pdf', function(err, document) {
+	console.log(err);
+	notifyClient.sendEmail('your_notify_api_key', 'recipient@email.com', {
+		personalisation: notifyClient.prepareUpload(document),
```
Comment (leohemsted) [inline README.md]: same comment as DOCUMENTATION.md

File: client/notification.js (hunk 1 above)
Comment (leohemsted) [reply client/notification.js]: do we do this check when sending precompiled? (if not, should we?)

Comment (servingUpAces) [reply client/notification.js]: No we don't. We probably should, I'm not sure what that size limit is, unless we use the limit we set on the size for a request, which is 5MB

File: DOCUMENTATION.md (hunk 3 above)
Comment (CrystalPea) [reply DOCUMENTATION.md]: Ahh, good call, I will change it.

File: client/notification.js (hunk 1 above)
Comment (CrystalPea) [reply client/notification.js]: Should I set size limit for precompiled to 5MB?

**Comment by CrystalPea** [comment]
@servingUpAces integration test added :)

File: DOCUMENTATION.md [hunk 5]
```diff
@@ -250,6 +250,79 @@ If you omit this argument your default email reply-to address will be set for th
...
+|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Document didn't pass the virus scan"`<br>`}]`|The document you upload must be virus free|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Error: Your system clock must be accurate to within 30 seconds"`<br>`}]`|Check your system clock|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Invalid token: signature, api token not found"`<br>`}]`|Use the correct type of [API key](#api-keys)|
+|`429`|`[{`<br>`"error": "RateLimitError",`<br>`"message": "Exceeded rate limit for key type TEAM/TEST/LIVE of 3000 requests per 60 seconds"`<br>`}]`|Refer to [API rate limits](#api-rate-limits) for more information|
+|`429`|`[{`<br>`"error": "TooManyRequestsError",`<br>`"message": "Exceeded send limits (LIMIT NUMBER) for today"`<br>`}]`|Refer to [service limits](#service-limits) for the limit number|
+|`500`|`[{`<br>`"error": "Exception",`<br>`"message": "Internal server error"`<br>`}]`|Notify was unable to process the request, resend your notification.|
+
```
Comment (servingUpAces) [inline DOCUMENTATION.md]: Can you add the file too big error to this one too please?

**Pull request merged**